sdk/attestation/azure-security-attestation/inc/azure/attestation/attestation_client.hpp

```diff
@@ -0,0 +1,410 @@
+     * @brief Returns the version the client was configured with.
+     *
+     * @returns The API version used when communicating with the attestation service.
+     */
+    std::string ClientVersion() const;
+    /**
+     * @brief Construct a new Key Client object from another key client.
+     *
+     * @param attestationClient An existing attestation client.
+     */
+    explicit AttestationClient(AttestationClient const& attestationClient)
+        : m_endpoint(attestationClient.m_endpoint), m_apiVersion(attestationClient.m_apiVersion),
+          m_pipeline(attestationClient.m_pipeline){};
+
+    /**
+     * Retrieves metadata about the attestation signing keys in use by the attestation service.
+     *
+     * Retrieve the OpenID metadata for this attestation service instance..
+     *
+     * @return an {@link AttestationOpenIdMetadata} containing metadata about the specified service
+     * instance.
+     */
+    Response<Models::AttestationOpenIdMetadata> GetOpenIdMetadata(
+        Azure::Core::Context const& context = Azure::Core::Context::ApplicationContext) const;
+
+    /**
+     * @brief Retrieve the attestation signing certificates for this attestation instance.
+     *
+     * @returns Attestation Metadata.
+     */
+    Response<std::vector<Models::AttestationSigner>> GetAttestationSigningCertificates(
+        Azure::Core::Context const& context = Azure::Core::Context::ApplicationContext) const;
+
+    /**
+     * @brief Attest an SGX enclave, returning an attestation token representing the result
+     * of the attestation operation.
+     *
+     * @param sgxQuoteToAttest - SGX Quote to be validated by the attestation service.
+     * @param options - Options to the attestation request (runtime data, inittime data, etc).
+     * @param context - Context for the operation.
+     *
+     * @returns Response<{@link AttestationToken}<{@link AttestationResult}>> - The result of the
+     * attestation operation
+     */
+    Response<AttestationToken<Models::AttestationResult>> AttestSgxEnclave(
+        std::vector<uint8_t> const& sgxQuoteToAttest,
+        AttestOptions options = AttestOptions(),
+        Azure::Core::Context const& context = Azure::Core::Context::ApplicationContext) const;
+
+    /**
+     * @brief Attest an OpenEnclave report, returning an attestation token representing the result
+     * of the attestation operation.
+     *
+     * @param openEnclaveReportToAttest - OpenEnclave Report to be validated by the attestation
+     * service.
+     * @param options - Options to the attestation request (runtime data, inittime data, etc).
+     * @param context - Context for the operation.
+     *
+     * @returns Response<AttestationToken<AttestationResult>> - The result of the attestation
+     * operation
+     */
+    Response<AttestationToken<Models::AttestationResult>> AttestOpenEnclave(
+        std::vector<uint8_t> const& openEnclaveReportToAttest,
+        AttestOptions options = AttestOptions(),
```

Comment:
might want to consider having operation-options unique per operation.  Right now, `OpenEnclave` and `SgxEnclave` are getting marry forever :)